### PR TITLE
Anchor now follow proxy pattern

### LINKED ISSUE #71: Convert Anchor contract to support proxy pattern
In order to save deployment gas, Anchor contract should support proxy pattern.

### PR DESCRIPTION
**Pull request by 0xsarvesh**
fixes #71

### CODE REVIEW

File: contracts/axiom/Axiom.sol [hunk 1]
```diff
@@ -47,6 +47,13 @@ contract Axiom is AxiomI, ProxyFactory, ConsensusModule {
         )
     );
 
+    /** The callprefix of the Anchor::setup function. */
```
Comment (benjaminbollen) [inline contracts/axiom/Axiom.sol]: please detail in comments the setup function for easy comparison

**Review by benjaminbollen — DISMISSED** [review]
LGTM

File: contracts/lib/CircularBufferUint.sol [hunk 2]
```diff
@@ -41,15 +41,15 @@ contract CircularBufferUint {
...
     /**
-     * @notice Create a new buffer with the size `_maxItems`.
+     * @notice Setup buffer with the size `_maxItems`.
      *
      * @param _maxItems Defines how many items this buffer stores before
      *                  overwriting older items.
      */
-    constructor(uint256 _maxItems) public {
+    function setupCircularBuffer(uint256 _maxItems) internal {
+
```
Comment (deepesh-kn) [inline contracts/lib/CircularBufferUint.sol:52]: `setupCircularBuffer` should be allowed to call only once.

**Review by deepesh-kn — DISMISSED** [review]
LGTM, just one comment.

File: contracts/lib/CircularBufferUint.sol (hunk 2 above)
Comment (0xsarvesh) [reply contracts/lib/CircularBufferUint.sol:52]: Done

**Review by deepesh-kn — APPROVED** [review]
LGTM 👍

**Pull request merged**